File: core/internal/models/file_info.go
package models

import "time"

// FileInfo 表示文件或目录的基本信息
type FileInfo struct {
	Name    string    `json:"name"`
	Path    string    `json:"path"`
	Size    int64     `json:"size"`
	Mode    string    `json:"mode"`
	ModTime time.Time `json:"modTime"`
	IsDir   bool      `json:"isDir"`
	Owner   string    `json:"owner"`
	Group   string    `json:"group"`
}

File: src/types/FileInfo.ts
export interface FileInfo {
    name: string
    path: string
    size: number
    isDir: boolean
    mode: string
    modTime: string
    owner: string
    group: string
    isSymlink: boolean
    linkTarget: string
}

export type SortBy = 'name' | 'size' | 'modTime'
export type SortOrder = 'asc' | 'desc'